editor/src/editor/ParametersWindow.cpp

```diff
@@ -0,0 +1,101 @@
+//
+// Created by macie on 10/08/2021.
+//
+
+#include <R3E/system/Config.h>
+
+#include <editor/ParametersWindow.h>
+
+#include <Editor.h>
+
+
+using namespace editor;
+
+ParametersWindow::ParametersWindow(tgui::Gui* gui, tgui::Theme* theme) :
+        ChildWindow(gui, theme, "Parameters editor",
+                    (sf::Vector2f(CONF<int>("window_width_px"), CONF<int>("window_height_px")) -
+                     CONF<sf::Vector2f>("popup_window_size")) / 2.0f,
+                    CONF<sf::Vector2f>("popup_window_size"),
+                    "parameters_window")
+{
+    child_->setTitle("Map parameters");
+
+    grid_ = tgui::Grid::create();
+    grid_->setPosition("50% - width/2", "20");
+    grid_->setSize("90%", "80%");
+    grid_->setAutoSize(true);
+    child_->add(grid_);
+
+    auto label = tgui::Label::create();
+    label->setRenderer(theme_->getRenderer("Label"));
+    label->setText("Shader:");
+    label->setTextSize(CONF<float>("label_text_size"));
+
+    grid_->addWidget(label, 0, 0);
+
+    shader_box_ = tgui::EditBox::create();
+    shader_box_->setRenderer(theme_->getRenderer("EditBox"));
+    shader_box_->setSize("35%", CONF<float>("text_box_height"));

```

Comment:
shader_box_, background_box_, and etc... combine into one `std::vector<tgui::EditBox::Ptr>`.

Then create it in a for loop like this:
```
for (const auto& param : implemented_params_) {
...
}
```

implemented_params_ should be initialized with {"shader", "background_color", "lightning_color"}.
Then inside setParameters it should be also coded more generically.